src/point-cloud-layer.ts

```diff
@@ -0,0 +1,188 @@
+const {
+  data: _data,
+  getPosition: _getPosition,
+  ..._upstreamDefaultProps
+} = PointCloudLayer.defaultProps;
+
+// Default props added by us
+const ourDefaultProps = {
+  _validate: true,
+};
+
+// @ts-expect-error Type error in merging default props with ours
+const defaultProps: DefaultProps<GeoArrowPointCloudLayerProps> = {
+  ..._upstreamDefaultProps,
+  ...ourDefaultProps,
+};
+
+export class GeoArrowPointCloudLayer<
+  ExtraProps extends {} = {},
+> extends CompositeLayer<GeoArrowPointCloudLayerProps & ExtraProps> {
+  static defaultProps = defaultProps;
+  static layerName = "GeoArrowPointCloudLayer";
+
+  getPickingInfo(
+    params: GetPickingInfoParams & {
+      sourceLayer: { props: GeoArrowExtraPickingProps };
+    },
+  ): GeoArrowPickingInfo {
+    return getPickingInfo(params, this.props.data);
+  }
+
+  renderLayers(): Layer<{}> | LayersList | null {
+    const { data: table } = this.props;
+
+    const pointVector = getGeometryVector(table, EXTENSION_NAME.POINT);
+    if (pointVector !== null) {
+      return this._renderLayersPoint(pointVector);
+    }
+
+    const geometryColumn = this.props.getPosition;
+    if (
+      geometryColumn !== undefined &&
+      ga.vector.isPointVector(geometryColumn)
+    ) {
+      return this._renderLayersPoint(geometryColumn);
+    }
+
+    throw new Error("geometryColumn not GeoArrow point");
+  }
+
+  _renderLayersPoint(
+    geometryColumn: ga.vector.PointVector,
+  ): Layer<{}> | LayersList | null {
+    const { data: table } = this.props;
+
+    if (this.props._validate) {
+      assert(
+        ga.vector.isPointVector(geometryColumn),
+        "The geometry column is not a valid PointVector.",
+      );
+      assert(
+        geometryColumn.type.listSize === 3,
+        "Points of a PointCloudLayer in the geometry column must be three-dimensional.",
+      );
+      validateAccessors(this.props, table);
+    }
+
+    // Exclude manually-set accessors
+    const [accessors, otherProps] = extractAccessorsFromProps(this.props, [
+      "getPosition",
+    ]);
+    const tableOffsets = computeChunkOffsets(table.data);
+
+    const layers: PointCloudLayer[] = [];
+    for (
+      let recordBatchIdx = 0;
+      recordBatchIdx < table.batches.length;
+      recordBatchIdx++
+    ) {
+      const geometryData = geometryColumn.data[recordBatchIdx];
+      const flatCoordsData = ga.child.getPointChild(geometryData);
+      const flatCoordinateArray = flatCoordsData.values;
+
+      const props: PointCloudLayerProps = {
+        // Note: because this is a composite layer and not doing the rendering
+        // itself, we still have to pass in our defaultProps
+        ...ourDefaultProps,
+        ...otherProps,
+
+        // @ts-expect-error used for picking purposes
+        recordBatchIdx,
+        tableOffsets,
+
+        id: `${this.props.id}-geoarrow-pointcloud-${recordBatchIdx}`,
+        data: {
+          length: geometryData.length,
```

Comment:
You need to pass in a `data` argument here as well, otherwise function accessors won't work. See for example the scatterplot layer:
https://github.com/geoarrow/deck.gl-layers/blob/4887ce8f9e4fa02b1a53515e4b60ec83c63b00be/src/scatterplot-layer.ts#L178-L179